RabbitMQ.Stream.Client/Reliable/Producer.cs

```diff
@@ -232,7 +268,7 @@ public async ValueTask Send(Message message)
         }
         finally
         {
-            SemaphoreSlim.Release();
+            // SemaphoreSlim.Release();
         }
```

Comment:
If we are not locking the semaphore in this method, can we delete this `finally` block?